modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/VolatilePageMemoryMvPartitionStorage.java

```diff
@@ -149,17 +153,32 @@ public void lastAppliedOnRebalance(long lastAppliedIndex, long lastAppliedTerm)
     }
 
     /**
-     * Destroys internal structures backing this partition.
+     * Destroys internal structures (including indices) backing this partition.
      *
-     * @return future that completes when the destruction completes.
+     * @param removeIndexDescriptors Whether indices should be completely removed, not just their contents destroyed.
      */
-    public CompletableFuture<Void> destroyStructures() {
-        // TODO: IGNITE-18531 - destroy indices.
+    public void destroyStructures(boolean removeIndexDescriptors) {
```

Comment:
I would propose to make this method "private" and introduce two public methods instead

Reply:
Done